mergin/client.py

```diff
@@ -737,3 +737,26 @@ def get_file_diff(self, project_dir, file_path, output_diff, version_from, versi
         job = download_diffs_async(self, project_dir, file_path, version_from, version_to)
         pull_project_wait(job)
         download_diffs_finalize(job, output_diff)
+
```

Comment:
I think we should add some checks to pull_project_async() and push_project_async() to deal with unfinished syncs:
- in push_project_async() we should raise ClientError if there's unfinished pull
- in pull_project_async(), in case of unfinished pull we should first try to resolve it, and if that fails we would raise ClientError

Also worth checking behavior of those in the auto test...

Reply:
Added.